### PR TITLE
New pip resolver

### PR DESCRIPTION
**Pull request by flying-sheep**
@ivirshup why Ubuntu 16.04? Isn’t that super old?

### CODE REVIEW

File: scanpy/__init__.py [hunk 1]
```diff
@@ -22,6 +22,7 @@
 
 # has to be done at the end, after everything has been imported
 import sys
+sys.modules.update({f'{__name__}.{m}': globals()[m] for m in ['tl', 'pp', 'pl']})
```
Comment (ivirshup) [inline scanpy/__init__.py:25]: Why was this added?

Comment (flying-sheep) [reply scanpy/__init__.py:25]: This makes it so that `sc.tl` and so on is a real module python knows about, instead of a global variable that happens to point to a module object. Sphinx 3.2 got more strict about where it allows imports from. It won’t use attribute access for something we claim is a module, it’ll import it as a module.

Also people can now do `from scanpy.tl import ...` which is consistent to what we claim in our modules (I.e. that `sc.tl` is a module)

Comment (ivirshup) [reply scanpy/__init__.py:25]: Is this better than having a `tl.py` which has: `from .tools import *`?

**Comment by ivirshup** [comment]
AFAICT `ubuntu 16.04` is what they use in most of their examples. Considering many of our users will be on academic clusters, I think old-ish versions of linux are reasonable to test against. The alternative would probably be `ubuntu-18.04`, which we should switch to when `16` is out of support.

[Here are the options](https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/hosted?view=azure-devops&tabs=yaml#software).

File: scanpy/__init__.py (hunk 1 above)
Comment (flying-sheep) [reply scanpy/__init__.py:25]: I think so. That way the modules are actually identical and not two different modules containing references to the same objects.

**Comment by flying-sheep** [comment]
We should have used the newer LTS from the start, 16.04 is out of support in 5 months.
